File: python/design-patterns/template/naive/naive_beverages.py
class CoffeeMaker:
    def prepare(self):
        print("Boiling water") # ❌ duplicated
        print("Brewing coffee")
        print("Pouring into cup") # ❌ duplicated
        print("Adding sugar")

class TeaMaker:
    def prepare(self):
        print("Boiling water") # ❌ duplicated
        print("Steeping tea")
        print("Pouring into cup") # ❌ duplicated
        print("Adding lemon")

File: typescript/design-patterns/template/naive/naive-beverages.ts
// Naive approach: separate classes replicate identical boiling/pouring steps
class CoffeeMaker {
  prepare() {
    console.log("Boiling water..."); // ❌ duplicated boilerplate
    console.log("Dripping coffee through filter...");
    console.log("Pouring into cup...");    // ❌ duplicated boilerplate
    console.log("Adding milk and sugar...");
  }
}

class TeaMaker {
  prepare() {
    console.log("Boiling water..."); // ❌ duplicated boilerplate
    console.log("Steeping the tea...");
    console.log("Pouring into cup...");    // ❌ duplicated boilerplate
    console.log("Adding lemon...");
  }
}
// Changing the boiling temperature means editing every single preparation class!